### PR TITLE
koordetector: init interference detection related crds

### PR DESCRIPTION
**Pull request by songtao98**
Signed-off-by: songtao98 <songtao2603060@gmail.com>

### Ⅰ. Describe what this PR does
This PR initialize `intereference detection` related crds' type definition using kubebuilder. The detailed `Spec` and `Status` information are **NOT** implemented yet. 

In future PRs, we will update these crds with details step by step. Basic information about these CRDs can be referenced as below: 

- `InterferenceDetectionRule`: Every workload has its own rules for interference detection feature. Every rule includes strategies for each metric. Strategy includes `Enable`, `Threshold`, `Metric range`, etc. 

- `InterferenceMetricCheckpoint`: Interference manager collects interference metrics from datasources(currently Prometheus) and aggregate those metrics into `Histograms`. This process is done in memory and the result `Histogram` needs to be stored into this CRD one by one checkpoint. 

Also notice that the code structure of Koordetector is not defined yet, too. For example controller logics might be implemented in `pkg` directory which is consistent with Koordinator. 

### Ⅳ. Special notes for reviews
This is the very beginning version of Koordetector's code structure, following kubebuilder's default generated structure. It will be optimized in future PRs. 

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/interference/v1alpha1/interferencebase_types.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	OtherLabels map[string]string `json:"otherLabels,omitempty"`
+}
+
+// InterferenceBaseSpec defines the desired state of InterferenceBase
+type InterferenceBaseSpec struct {
+	BaseInfo *CheckpointMeta `json:"checkpointInfo,omitempty"`
+}
+
+type Baseline struct {
+	Miu   string `json:"miu,omitempty"`
```
Comment (songtao98) [inline apis/interference/v1alpha1/interferencebase_types.go]: maybe use resource.Quantity to represent float64, string is ok

File: api/v1alpha1/interferencedetectionrule_types.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+type InterferenceDetectionRuleStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// InterferenceDetectionRule is the Schema for the interferencedetectionrules API
+type InterferenceDetectionRule struct {
```
Comment (zwzhang0107) [inline api/v1alpha1/interferencedetectionrule_types.go]: more comments need for the purpose of InterferenceDetectionRule and InterferenceDetectionCheckpoint

File: api/v1alpha1/interferencebase_types.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// InterferenceBaseSpec defines the desired state of InterferenceBase
+type InterferenceBaseSpec struct {
```
Comment (zwzhang0107) [inline api/v1alpha1/interferencebase_types.go]: can be merged as the status of Rule

File: main.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
```
Comment (zwzhang0107) [inline main.go]: controller/main pkg should be adjuested as
cmd/controller/main.go
cmd/agent/main.go
pkg/controllers/
...

Comment (songtao98) [reply main.go]: sure, also mv dockerfile to docker directory

**Pull request merged**